src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -77,9 +79,7 @@ public InventorySync(@Nonnull final InventorySyncOptions syncOptions) {
 
     /**
      * Iterates through the whole {@code inventories} list and accumulates its valid drafts to batches. Every batch
-     * is then processed by {@link InventorySync#processBatch(List)}. For invalid drafts from {@code inventories}
-     * "processed" and "failed" counters from statistics are incremented and error callback is executed. A valid draft
-     * is a {@link InventoryEntryDraft} object that is not {@code null} and its SKU is not empty.
+     * is then processed by {@link InventorySync#processBatch(List)}.
```

Comment:
Yes it is a side effect but you couldn't tell that looking at the method body, could you? I found a  javadoc too broad. It describes more than the method actually do, so it generates a redundancy, and makes that javadoc could become out of date if something has changed in dependent functions.

> "processed" and "failed" counters from statistics are incremented and error callback is executed

This is what the `handleError` method do, so I think it is enough to keep this description in a javadoc of `handleError`.